sdk/data/azcosmos/emulator_cosmos_gem_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package azcosmos
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGlobalEndpointManager(t *testing.T) {
+	emulatorTests := newEmulatorTests(t)
+	client := emulatorTests.getClient(t)
+
+	_, err := url.Parse(client.Endpoint())
+	if err != nil {
+		fmt.Println("Unable to parse default endpoint URI")
+		os.Exit(1)
+	}
+
+	preferredRegions := []string{"West US", "Central US"}
+	gem, err := NewGlobalEndpointManager(client, preferredRegions)
+	if err != nil {
+		t.Fatalf("failed to create globalEndpointManager: %v", err)
+	}
+	fmt.Printf("gem: %v", gem)
+
+	t.Run("TestGetAccountProperties", func(t *testing.T) {
+		accountProperties, err := gem.GetAccountProperties()
+		assert.NoError(t, err, "GetAccountProperties should not return an error")
+		assert.NotNil(t, accountProperties, "accountProperties should not be nil")
+
+		accountProperties.EnableMultipleWriteLocations = true
+		assert.Equal(t, accountProperties.EnableMultipleWriteLocations, true, "EnableMultipleWriteLocations should be true")
```

Comment:
This is testing a flag that was set on the previous line, what is the case where this might fail?